### PR TITLE
[Feature] Implement Currency formatter utility

### PR DESCRIPTION
**Pull request by GaneshSolGuruz**
# Pull Request

## Description

- Implemented a CurrencyFormatter utility to handle numbers with symbols, decimal digits, compact formats, and custom grouping/decimal separators.
- Created a CurrencyFormatExtensions to directly format amounts using the current app locale.
- Implemented fallback locale handling to avoid errors when the locale is missing or invalid.
- Add unit test cases covering different locales (USD, INR), numeric types, edge cases, and formatting options.
- For now currency formatter utility used in home and product detail screen to showing price.
- **Note: Currency conversion will be added soon.**
-  **Note: The formatter currently supports only formatting.**

## Type of Change

- [ ] Bug fix
- [x] New feature
- [x] Code refactoring
- [x] Test cases
- [ ] Documentation update
- [ ] UI/UX improvement
- [ ] Performance improvement
- [ ] Build configuration change
- [ ] Other (please describe):

## Checklist
- [x] Before requesting for review, I have self reviewed all the changes in this PR
- [x] My code follows the project's coding style guidelines
- [x] I have added any necessary tests
- [x] New and existing tests pass locally with my changes
- [x] Github pipeline passes with my changes

## Additional Notes

**> NOTE: The formatter currently supports only formatting.**
**> NOTE: Currency conversion will be added soon.**

1. Update ProductDetail golden images for Linux and macOS due to replacing empty space with WidgetSpan after implementing currency formatter.

2. We use the fixed number sample "100" num to detect currency symbol position in _getCurrencySymbolPosition() method.

// Why 100?
 - It is guaranteed to produce digits ("100") across all locales.
 - It avoids edge cases like decimals, rounding, or scientific notation.
 - It makes it easy to check whether the symbol appears before or after the number.

 Why NOT use the actual dynamic value?
- The real amount may be small (e.g., 0.5) or large (e.g., 123456789), causing
 different formatting behaviours across locales.
 - It may include decimal points or grouping, making detection unreliable.
 - Using a fixed, predictable number ensures consistent and accurate symbol-position detection.

### CODE REVIEW

File: lib/utils/extensions/currency_formatter_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:skelter/i18n/app_localizations.dart';
+import 'package:skelter/utils/currency_formatter.dart';
+
+extension CurrencyFormatterExtensions on BuildContext {
```
Comment (mihirjsolguruz) [inline lib/utils/extensions/currency_formatter_extensions.dart]: move this to `build_context_ext.dart` since it's part of BuildContext

File: lib/utils/currency_formatter.dart [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  /// [groupingSeparator] allows customizing the grouping separator
+  /// (e.g., ',' -> 1,000).
+  /// [decimalSeparator] allows customizing the decimal separator
+  /// (e.g., '.' -> 10.50).
+  /// [symbolSeparator] allows adding a separator between the symbol and the
+  /// amount (e.g., ' ' -> $ 100).
+  /// [compactFormatType] specifies the type of compact formatting to apply
+  /// (e.g., [CompactFormatType.short] -> $1.2K).
+  /// [fallbackValue] is returned if the amount is null or invalid
+  /// (default is '0.00').
```
Comment (mihirjsolguruz) [inline lib/utils/currency_formatter.dart]: did we covered all these in test cases?

**Review by mihirjsolguruz — APPROVED** [review]
